src/libraries/Microsoft.Extensions.Logging.Console/src/SystemdConsoleFormatter.cs

```diff
@@ -0,0 +1,174 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class SystemdConsoleFormatter : ConsoleFormatter, IDisposable
+    {
+        [ThreadStatic]
+        private static StringBuilder _logBuilder;
+        private IDisposable _optionsReloadToken;
+
+        private const string LoglevelPadding = ": ";
+        private static readonly string _messagePadding = new string(' ', GetSyslogSeverityString(LogLevel.Information).Length + LoglevelPadding.Length);
+
+        public SystemdConsoleFormatter(IOptionsMonitor<ConsoleFormatterOptions> options)
+            : base(ConsoleFormatterNames.Systemd)
+        {
+            FormatterOptions = options.CurrentValue;
+            ReloadLoggerOptions(options.CurrentValue);
+            _optionsReloadToken = options.OnChange(ReloadLoggerOptions);
+        }
+
+        private void ReloadLoggerOptions(ConsoleFormatterOptions options)
+        {
+            FormatterOptions = options;
+        }
+
+        public void Dispose()
+        {
+            _optionsReloadToken?.Dispose();
+        }
+
+        internal ConsoleFormatterOptions FormatterOptions { get; set; }
+
+        public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeProvider scopeProvider, TextWriter textWriter)
+        {
+            LogLevel logLevel = logEntry.LogLevel;
+            string category = logEntry.Category;
+            int eventId = logEntry.EventId.Id;
+            Exception exception = logEntry.Exception;
+            string message = logEntry.Formatter(logEntry.State, exception);
+
+            StringBuilder logBuilder = _logBuilder;
```

Comment:
We shouldn't need this local `StringBuilder` at all. We should just write to the `textWriter` directly. We are allocating unnecessary intermediate strings this way.